src/main/java/miku/parser/CommandMikuParser.java

```diff
@@ -0,0 +1,72 @@
+package miku.parser;
+
+import miku.command.AddCommand;
+import miku.command.Command;
+import miku.command.ExitCommand;
+import miku.command.FindCommand;
+import miku.command.MarkCommand;
+import miku.command.RemoveCommand;
+import miku.command.ShowListCommand;
+import miku.command.UnmarkCommand;
+import miku.exception.InvalidCommandException;
+import miku.task.Deadline;
+import miku.task.Event;
+import miku.task.Todo;
+
+/**
+ * Parses the input command strings
+ */
+public class CommandMikuParser extends MikuParser {
+    private static String regexMark = "mark \\d+";
+    private static String regexUnmark = "unmark \\d+";
+    private static String regexToDo = "todo .*";
+    private static String regexDeadline = "deadline .* /by .*";
+    private static String regexEvent = "event .* /.* /.*";
+    private static String regexRemove = "delete \\d+";
+    private static String regexFind = "find .*";
```

Comment:
This is really cool! I didn't know you could do this. I split mine up one at a time, e.g (event) input.split.("/from").split("/to") etc

Reply:
Woah yeah i agree, I was shocked when I saw this! Really cool!